src/TLuaInterpreter.cpp

```diff
@@ -10244,72 +9819,70 @@ int TLuaInterpreter::setDoor(lua_State* L)
-                  || pR->getOtherMap().values().contains(QStringLiteral("1%1").arg(exitCmd)))) {
-                // And NOT a special one either
-                lua_pushnil(L);
-                lua_pushfstring(L,
-                                "setDoor: bad argument #2 value (room with id %d does not have a special\n"
-                                "exit in direction \"%s\".)",
-                                roomId,
-                                exitCmd.toUtf8().constData());
-                return 2;
-            }
-            // else IS a valid special exit - so fall out of if and continue
-        } else {
-            // Is a normal exit so see if it is valid
-            if (!(((!exitCmd.compare(QStringLiteral("n"))) && (pR->getExit(DIR_NORTH) > 0 || pR->exitStubs.contains(DIR_NORTH)))
-                  || ((!exitCmd.compare(QStringLiteral("e"))) && (pR->getExit(DIR_EAST) > 0 || pR->exitStubs.contains(DIR_EAST)))
-                  || ((!exitCmd.compare(QStringLiteral("s"))) && (pR->getExit(DIR_SOUTH) > 0 || pR->exitStubs.contains(DIR_SOUTH)))
-                  || ((!exitCmd.compare(QStringLiteral("w"))) && (pR->getExit(DIR_WEST) > 0 || pR->exitStubs.contains(DIR_WEST)))
-                  || ((!exitCmd.compare(QStringLiteral("ne"))) && (pR->getExit(DIR_NORTHEAST) > 0 || pR->exitStubs.contains(DIR_NORTHEAST)))
-                  || ((!exitCmd.compare(QStringLiteral("se"))) && (pR->getExit(DIR_SOUTHEAST) > 0 || pR->exitStubs.contains(DIR_SOUTHEAST)))
-                  || ((!exitCmd.compare(QStringLiteral("sw"))) && (pR->getExit(DIR_SOUTHWEST) > 0 || pR->exitStubs.contains(DIR_SOUTHWEST)))
-                  || ((!exitCmd.compare(QStringLiteral("nw"))) && (pR->getExit(DIR_NORTHWEST) > 0 || pR->exitStubs.contains(DIR_NORTHWEST)))
-                  || ((!exitCmd.compare(QStringLiteral("up"))) && (pR->getExit(DIR_UP) > 0 || pR->exitStubs.contains(DIR_UP)))
-                  || ((!exitCmd.compare(QStringLiteral("down"))) && (pR->getExit(DIR_DOWN) > 0 || pR->exitStubs.contains(DIR_DOWN)))
-                  || ((!exitCmd.compare(QStringLiteral("in"))) && (pR->getExit(DIR_IN) > 0 || pR->exitStubs.contains(DIR_IN)))
-                  || ((!exitCmd.compare(QStringLiteral("out"))) && (pR->getExit(DIR_OUT) > 0 || pR->exitStubs.contains(DIR_OUT))))) {
-                // No there IS NOT a stub or real exit in the exitCmd direction
-                lua_pushnil(L);
-                lua_pushfstring(L,
-                                "setDoor: bad argument #2 value (room with id %d does not have a normal exit\n"
-                                "or a stub exit in direction \"%s\".)",
-                                roomId,
-                                exitCmd.toUtf8().constData());
-                return 2;
-            }
-            // else IS a valid stub or real normal exit -fall through to continue
+    }
+    exitCmd = QString::fromUtf8(lua_tostring(L, 2));
```

Comment:
```suggestion
    QString exitCmd = QString::fromUtf8(lua_tostring(L, 2));
```
and drop the line above